xcodeproj/internal/xcodeprojinfos.bzl

```diff
@@ -57,37 +72,22 @@ _TOOLS_REPOS = {
 }
 
 # Just a slight optimization to not process things we know don't need to have
-# out provider.
+# our provider
 def _should_create_provider(*, ctx, target):
     if ctx.rule.kind in _INTERNAL_RULE_KINDS:
         return False
-    if not target.label.workspace_name:
-        return True
     if BuildSettingInfo in target:
         return False
+    if not target.label.workspace_name:
+        return True
```

Comment:
This is a slight behavior change, mainly preventing processing of Starlark build flags/settings in the main repo as well as external ones. Should only be a perf boost.

I'll pull it out to its own PR though.